src/binder/clrprivbindercoreclr.cpp

```diff
@@ -159,8 +159,13 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 {
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        SString tpaAssemblyPath;
+                        SString assemblyPath;
+                        BinderGetImagePath(pPEImage, assemblyPath);
+                        BinderGetImagePath(pCoreCLRFoundAssembly->GetPEImage(), tpaAssemblyPath);
+                        if(!tpaAssemblyPath.EqualsCaseInsensitive(assemblyPath))
+                            // If we were able to bind to a TPA assembly, then fail the load
+                            IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
```

Comment:
also with the exception you get very generic message string. 
Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
I am planning to change it to following
Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Assembly with same identity found at standard probing locations.

what do you think?